ocaml/database/database_test.ml

```diff
@@ -626,6 +627,32 @@ functor
           "read_field_where <valid table> <valid return> <valid field> <valid \
            value>" ;
       test_invalid_where_record "read_field_where" (Client.read_field_where t) ;
+
+      (* before changing the UUID, the new UUID should be missing *)
+      expect_missing_uuid "VM" new_uuid (fun () ->
+          let (_ : string) = Client.db_get_by_uuid t "VM" new_uuid in
+          ()
+      ) ;
+      (* change UUID, can happen during VM import *)
```

Comment:
Another frequent operation: VM.revert changes uuids of VDIs

Reply:
Yes, I'm surprised there were 0 tests for this in XAPI's unit tests. I think we need some better testing for the fundamentals, maybe expand the quickcheck-like testing that we have in some places.
Sure XenRT will catch the most obvious ones, but we should be able to catch DB bugs earlier.

(well now there is 1 hand written test for 1 attribute)